CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc

```diff
@@ -0,0 +1,248 @@
+// -*- C++ -*-
+//
+// Package:    CondFormats/PCLConfig
+// Class:      AlignPCLThresholdsWriter
+// 
+/**\class AlignPCLThresholdsWriter AlignPCLThresholdsWriter.cc CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
+
+ Description: class to build the SiPixelAli PCL thresholds
+
+*/
+//
+// Original Author:  Marco Musich
+//         Created:  Wed, 22 Feb 2017 12:04:36 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/PCLConfig/interface/AlignPCLThresholds.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+//
+// class declaration
+//
+
+namespace DOFs {
+  enum dof{
+    X,
+    Y,
+    Z,
+    thetaX,
+    thetaY,
+    thetaZ,
+    extraDOF
+  };
+}
+
+class AlignPCLThresholdsWriter : public edm::one::EDAnalyzer<edm::one::SharedResources>  {
+   public:
+      explicit AlignPCLThresholdsWriter(const edm::ParameterSet&);
+      ~AlignPCLThresholdsWriter();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+   private:
+      virtual void beginJob() override;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+      virtual void endJob() override;
+      DOFs::dof mapOntoEnum(std::string coord);
+
+      // ----------member data ---------------------------
+      std::string m_record;
+      unsigned int m_minNrecords;
+      const std::vector<edm::ParameterSet> m_parameters;
+};
+
+//
+// constructors and destructor
+//
+AlignPCLThresholdsWriter::AlignPCLThresholdsWriter(const edm::ParameterSet& iConfig):
+  m_record(iConfig.getParameter<std::string>("record")),
+  m_minNrecords(iConfig.getParameter<unsigned int>("minNRecords")),
+  m_parameters(iConfig.getParameter<std::vector<edm::ParameterSet> >("thresholds"))
+{
+   //now do what ever initialization is needed
+   usesResource("TFileService");
```

Comment:
can you change that to `usesResource(TFileService::kSharedResource);`
it's technically the same, of course